updated notebooks to use latest langchain version

josefinarcasanova opened the pull request:
Content changes:

- Improved general writing
- Standardized environment setup sections
- Added my name to authors, as editor
- Added more references to LC docs

Code changes:

- Updated to latest langchain version
- Changed LLMChain and SimpleSequentialChain (deprecated) to LCEL Runnable flows

josefinarcasanova committed:
updated notebooks to use latest langchain version

krab7191 reviewed (changes requested):
Reviewing now. Can you just format the JSON so it's easier to see diffs and maintain ?

krab7191 reviewed (commented):
Let's also remove the cell output

krab7191 reviewed (commented):
I'm using python 3.13, looks like this causes an issue with pandas install. Lets make a note of that. https://stackoverflow.com/a/78719227/18236359

krab7191 commented on content/labs/Watsonx/WatsonxAI/files/203/LangChain_Integration.ipynb:
```diff
@@ -1,576 +1 @@
- "nbformat": 4,
- "nbformat_minor": 2
-}
+{"cells":[{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%% md\n"}},"source":["![image](https://raw.githubusercontent.com/IBM/watson-machine-learning-samples/master/cloud/notebooks/headers/watsonx-Prompt_Lab-Notebook.png)\n","\n","# Use Watsonx and `LangChain` to make a series of calls to a language model"]},{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%% md\n"}},"source":["## ...
```
Confirmed working locally with python3.11. See comment below. Let's just make a note in the notebook for users who want to run locally.

josefinarcasanova replied:
Done. Added environment setup explanation for wx a non-wx and considerations section to explain 3.13 non-support.

krab7191 commented on content/labs/Watsonx/WatsonxAI/files/203/LangChain_Memory.ipynb:
```diff
@@ -1,304 +1 @@
- "nbformat": 4,
- "nbformat_minor": 2
-}
+{"cells":[{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%% md\n"}},"source":["<a id=\"setup\"></a>\n","## LangChain memory with watsonx.ai models\n","\n","This notebook contains sample code for using memory buffers with models included in watsonx.ai.\n"]},{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%% md\n"}},"source":["### Install dependecies"]},{"cell_type":"code","executi ...
```
For memory we'll need another package: `!{sys.executable} -m pip install langchain-community | tail -n 1`

krab7191 replied:
Might want to look at formatting some of the output in this file too.

josefinarcasanova replied:
Done, added package and clarified as with the other notebooks. 

As in output you mean clearing it for export, or that doesn't display right?

krab7191 commented on content/labs/Watsonx/WatsonxAI/files/203/LangChain_Simple_RAG.ipynb:
```diff
@@ -1,253 +1 @@
- "nbformat": 4,
- "nbformat_minor": 2
-}
+{"cells":[{"cell_type":"markdown","metadata":{},"source":["![image](https://raw.githubusercontent.com/IBM/watson-machine-learning-samples/master/cloud/notebooks/headers/watsonx-Prompt_Lab-Notebook.png)\n","\n","# Implement a simple RAG use case with LangChain\n","\n","_Retrieval Augmented Generation (RAG)_ allows us to use LLMs to interact with \"external data\" i.e. data that was not used for mo ...
```
I think we need to change to `from langchain_community.embeddings import HuggingFaceEmbeddings` from `langchain_huggingface`

josefinarcasanova commented:
> Reviewing now. Can you just format the JSON so it's easier to see diffs and maintain ?

Sorry, not sure how to. Googled it quickly and didn't find a fix. Perhaps you know? I'm working the `ipynb` directly in VSCode.

josefinarcasanova committed:
fixed reviewed requests

- removed cell output
- added seperate setup for wx-ai and non-wx-ai envs (all notebooks)
- added Python 3.11 notes to integrations nb
- updated HuggingFaceEmbeddings import
- fixed typos
- deleted unused imports

josefinarcasanova committed:
updated lab to match new nb

- Updated theory to match new LangChain version
- Added references to new docs
- Deleted duplicated stuff, as the notebook is more explanatory now
- Deled unneeded images (deleted from doc)

josefinarcasanova committed:
fixed typo

krab7191 commented:
> > Reviewing now. Can you just format the JSON so it's easier to see diffs and maintain ?
> 
> Sorry, not sure how to. Googled it quickly and didn't find a fix. Perhaps you know? I'm working the `ipynb` directly in VSCode.

The way I do it in Vscode is use the command palette -> `view: reopen editor with` -> `text editor` and then format the json.

josefinarcasanova committed:
formatted json

josefinarcasanova commented:
> > > Reviewing now. Can you just format the JSON so it's easier to see diffs and maintain ?
> > 
> > 
> > Sorry, not sure how to. Googled it quickly and didn't find a fix. Perhaps you know? I'm working the `ipynb` directly in VSCode.
> 
> The way I do it in Vscode is use the command palette -> `view: reopen editor with` -> `text editor` and then format the json.

thank you for the instructions! I have formatted the JSON as requested.

Workshop is next week, would like to solve this PR before then. Are my changes ok? Something else I need to look into?
